internal/common/database/migrations_test.go

```diff
@@ -0,0 +1,46 @@
+package database
+
+import (
+	"testing"
+	"strconv"
+	"strings"
+	"io/fs"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReadMigrations(t *testing.T) {
+	mockFiles := []string{
+		"migration1.sql",
+	}
+	mockFS := createMockFileSystem(mockFiles)
+	migrations, err := ReadMigrations(mockFS, "path")
+
+	// Checks the suffix of files and handles file naming format
+	for _, migration := range migrations {
+		assert.True(t, strings.HasSuffix(migration.name, "migration.sql"), "Invalid file format for %v", migration.name)
+	}
+	
+	// Checks slice format
+	expectedMigrations := []Migration{
+		{id: 0, name: "", sql: ""},
+	}
+	assert.Len(t, migrations, 1, "Unexpected slice length")
+
+	// Slice first value should be int
```

Comment:
The comments for these assertions are nice, but probably not necessary - the code is very clear.